demo/header-footer.html

```diff
@@ -0,0 +1,99 @@
+<html>
+
+<head>
+  <title>Fallback element</title>
+  <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1, user-scalable=yes">
+
+  <style>
+    body {
+      font-family: Roboto, Helvetica, sans-serif;
+      margin: 0;
+    }
+
+    virtual-list {
+      width: 100vw;
+      height: 100vh;
+    }
+
+    header {
+      padding: 10px;
+      box-sizing: border-box;
+      width: 100%;
+      color: #333;
+      background-color: #ddd;
+    }
+  </style>
+</head>
+
+<body>
+
+  <virtual-list></virtual-list>
+
+  <script type="module">
+    import '../virtual-list-element.js';
+    import './contacts/contact-element.js';
+
+    const virtualList = document.querySelector('virtual-list');
+
+    const nodePool = { group: [], contact: [] };
```

Comment:
Add a comment here to explain that our items come in two shapes: contacts, or groups = `{ group: true, title, count }`. So, each of the following functions treats these cases different, and we have two different node pools for recycling among them.